### DISCUSSION TITLE
The receiver of channel(multi to multi broadcast) will hang after disconnect untill another endpoint post a new message

### DISCUSSION THREAD

**Discussion by kipade** [mutouyun/cpp-ipc#98] [General]
To reproduce:
1.  start first chat demo (mark c0)
2. start second chat demo (mark c1)
3. c1 post a message more than 59(>=60) characters
4. enter 'q' to c1 to make it being exit
Now, c1 will be blocked in recv and could not exit until another client post a new message. And further more, will c1 was blocking, c0 could not exit by enter 'q', too

**Comment by mutouyun** [comment]
It's probably a bug. I'll check it out on the weekend.

**Comment by mutouyun** [comment]
This is a stupid bug... See attached patch for modifications.
https://github.com/mutouyun/cpp-ipc/commit/6111722534b15bd49b944f1e9edb8e900df39e7d

I did not use Compiler Explorer to carefully analyze what happens with an uninitialized variable (yes, it's a ub) when gcc compiles this code.
However, the reason why this issue only occurs after 60 characters is related to the data_length configured in def.h.
I have submitted the fix to the master branch. You can pull it down to see if the issue still exists. Please let me know the results in a timely manner.

**Comment by kipade** [comment]
cool. It's OK now.
